crates/net/rpc-types/src/admin.rs

```diff
@@ -0,0 +1,72 @@
+use reth_primitives::{NodeRecord, U256};
+use serde::{Deserialize, Serialize};
+
+/// Represents the `admin_nodeInfo` response, which can be queried for all the information
+/// known about the running node at the networking granularity.
+///
+/// Note: this format is not standardized.  Reth follows geth's format,
+/// see: https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-admin
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NodeInfo {
+    /// Enode in URL format.
+    pub enode: String,
+    /// ID of the local node.
+    pub id: String,
+    /// IP of the local node.
+    pub ip: String,
+    /// Address exposed for listening for the local node.
+    #[serde(rename = "listenAddr")]
+    pub listen_addr: String,
+    /// Local node client name.
+    pub name: String,
```

Comment:
Seems that to properly fill this property we'd need to pull:
- The version + (stable/unstable)
- OS
- Rust version

Any pointers on how we should do so for this situation, or will we roll our own response here?